### PR TITLE
Added functionalities for conversion of different units

### PR DESCRIPTION
**Pull request by zaid-malek-simform**
Measurement Converter
This Measurement Converter have various functionality 
1. Length Converter: where we have different unit conversion options like Kilometer, Meter, Centimeter, Millimeter, Micrometer, Nanometer, Mile, Foot, Yard and Inch.
2. Area Converter: where we have different unit conversion options like Square Foot, Square meter, Square centimeter, Square millimeter, Square Kilometer, Square Inch and Square mile.
3. Volume Converter: where we have different unit conversion options like Litre, Cubic meter, Cubic centimeter and Cubic millimeter.
4. Weight Converter: where we have different unit conversion options like Kilogram. Gram, Pound, Ounce and Ton.
5. Speed Converter: where we have different unit conversion options.
6. Temperature Converter: where we have different unit conversion options like Celsius, Kelvin and Fahrenheit.
7. Number Converter: where we have different unit conversion options like Decimal,Binary,Hexa,Octal.
8. Currency Converter: where we have different Currency conversion options.

### CODE REVIEW

File: Length.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
+<?php
+if(isset($_POST['submit'])){
+  $opt1=$_POST['opt1'];
+  $opt2=$_POST['opt2'];
+  $val1=$_POST['val1'];
+  // $val2=$_POST['val2'];
```
Comment (dhaval-mistry-simform) [inline Length.php]: Remove this unwanted commented code.

File: Length.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  <!-- <link rel="stylesheet" href="css/demo.css">   -->
+  <script src="js/jquery.min.js"></script>
+
+</head>
+
+<body>
+  <!-- partial:index.partial.html -->
+
+
+  <div class="modal " id="lengthModal" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel"
```
Comment (dhaval-mistry-simform) [inline Length.php]: Remove extra space after model

File: area.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <title>Converter</title>
+  <!-- Bootstrap CSS -->
+  <link rel='stylesheet' href='https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.2/css/bootstrap.min.css'>
+  <!-- Font Awesome CSS -->
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css"/>
+  <!-- Style CSS -->
+  <link rel="stylesheet" href="css/style.css">
+  <!-- Demo CSS -->
+  <!-- <link rel="stylesheet" href="css/demo.css"> -->
```
Comment (dhaval-mistry-simform) [inline area.php]: Remove this unwanted code.

File: area.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  <script src="js/jquery.min.js"></script>
+
+</head>
+
+<body>
+  <!-- partial:index.partial.html -->
```
Comment (dhaval-mistry-simform) [inline area.php]: Remove this code.

File: conversion.php [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    "Mach" => 1255.08,
+    "Mlph" => 1.609344,
+    "Mps" => 3.6,
+    "Kmph" => 1,
+    "Kmps" => 3600,
+    "Knot" => 1.852
+);
+
+
+function len($val,$from,$to){
```
Comment (dhaval-mistry-simform) [inline conversion.php]: Provide function description and curly brackets must be in next line.

File: conversion.php [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    $meter=toMeter($val,$from);
```
Comment (dhaval-mistry-simform) [inline conversion.php]: Provide spacing between equal to and add space after like this $val, $from

File: conversion.php [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    echo fromMeter($meter,$to);
```
Comment (dhaval-mistry-simform) [inline conversion.php]: Add space after $meter,

File: conversion.php [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+    echo fromKMPH($kilo,$to);
+}
+function toKMPH($val,$from){
+    return $val*kmph[$from];
+}
+function fromKMPH($val,$to){
+    return $val/kmph[$to];
+}
+
+if(isset($_POST['type'])){
```
Comment (dhaval-mistry-simform) [inline conversion.php]: Make sure you do the changes for this file.

File: css/demo.css [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+   -webkit-font-smoothing: antialiased;
+   -moz-osx-font-smoothing: grayscale;
+   font-size: 15px;
+   background: #eee;
+}
+
+
+
+
+/* @media only screen and (min-width: 720px) {
```
Comment (dhaval-mistry-simform) [inline css/demo.css]: Remove this code

File: css/style.css [hunk 10]
```diff
@@ -0,0 +1,71 @@
+/* @media (min-width: 576px) {
+  .modal-dialog {
+    max-width: 400px;
+  }
+
+  .modal-dialog .modal-content {
+    padding: 1rem;
+  }
+} */
```
Comment (dhaval-mistry-simform) [inline css/style.css]: Remove this code.

File: css/style.css [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+  margin: -2rem 0rem 2rem;
+}
+
+.btn-round {
+  /* z-index: -2; */
+  width: 10rem;
+  height: 8rem;
+  border-radius: 1rem;
+  margin: 2rem;
+  /* flex-wrap: wrap; */
```
Comment (dhaval-mistry-simform) [inline css/style.css]: Remove this code.

File: css/style.css [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+  }
+} */
+
+
+.form-title {
+  margin: -2rem 0rem 2rem;
+}
+
+.btn-round {
+  /* z-index: -2; */
```
Comment (dhaval-mistry-simform) [inline css/style.css]: Remove this code.

File: css/style.css [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+* {
+  margin: 0;
+  padding: 0;
+}
+
+body {
+  font-family: 'Roboto', sans-serif;
+  font-style: normal;
+  font-weight: 300;
+  /* font-smoothing: antialiased; */
```
Comment (dhaval-mistry-simform) [inline css/style.css]: Remove this code.

File: css/style.css [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+
+
+
+/* @media only screen and (min-width: 720px) {
+   body {
+      max-width: 720px;
+      margin-left: auto;
+      margin-right: auto;
+      padding: 24px;
+   }
```
Comment (dhaval-mistry-simform) [inline css/style.css]: Remove this code.

File: css/style.css [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+.modal-dialog {
+  width: 70rem;
+}
+
+.modal-content {
+  border-radius: 10px;
+  padding-top: 2rem;
+  /* height: 40rem; */
```
Comment (dhaval-mistry-simform) [inline css/style.css]: Remove this code.

File: currency.php [hunk 16]
```diff
@@ -0,0 +1,113 @@
+<?php
+if(isset($_POST['submit'])){
+    $from=$_POST['opt1'];
+    $to=$_POST['opt2'];
+    $value=$_POST['val1'];
+    $req_url = "https://api.exchangerate-api.com/v4/latest/$from";
```
Comment (dhaval-mistry-simform) [inline currency.php]: Variable name should be camelCased.

File: currency.php [hunk 17]
```diff
@@ -0,0 +1,113 @@
+<?php
+if(isset($_POST['submit'])){
```
Comment (dhaval-mistry-simform) [inline currency.php]: Please format the document properly

File: currency.php [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <title>Converter</title>
+  <!-- Bootstrap CSS -->
+  <link rel='stylesheet' href='https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.2/css/bootstrap.min.css'>
+  <!-- Font Awesome CSS -->
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css"/>
+  <!-- Style CSS -->
+  <link rel="stylesheet" href="css/style.css">
+  <!-- Demo CSS -->
+  <!-- <link rel="stylesheet" href="css/demo.css"> -->
```
Comment (dhaval-mistry-simform) [inline currency.php]: Remove this code.

File: currency.php [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+  <script src="js/jquery.min.js"></script>
+
+</head>
+
+<body>
+  <!-- partial:index.partial.html -->
```
Comment (dhaval-mistry-simform) [inline currency.php]: Remove this code.

File: currency.php [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+      <div class="modal-content">
+
+        <div class="modal-body">
+          <div class="form-title text-center">
+            <h2><a href="index.php" class="text-dark"><i class="fa fa-home" aria-hidden="true"></i></a>  Volume Conversion</h2><hr>
+            
+          </div>
+          <div class="d-flex justify-content-center text-center">
+            <form action="" method="POST">
+                <!-- <input type="hidden" class="form-control" id="type" name="type" value="volume"> -->
```
Comment (dhaval-mistry-simform) [inline currency.php]: Remove this code.

File: index.php [hunk 21]
```diff
@@ -0,0 +1,85 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <title>Converter</title>
+  <!-- Bootstrap CSS -->
+  <link rel='stylesheet' href='https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.2/css/bootstrap.min.css'>
+  <!-- Font Awesome CSS -->
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css" integrity="sha512-SzlrxWUlpfuzQ+pcUCosxcglQRNAq/DZjVsC0lE40xsADsfeQoEypE+enwcOiGjk/bSuGGKHEyjSoQ1zVisanQ==" crossorigin="anonymous" referrerpolicy="no-referrer" />
+  <!-- Style CSS -->
+  <link rel="stylesheet" href="css/style.css">
+  <!-- Demo CSS -->
+  <!-- <link rel="stylesheet" href="css/demo.css"> -->
```
Comment (dhaval-mistry-simform) [inline index.php]: Remove this code.

File: index.php [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+          <h2><a href="index.php" class="text-dark"><i class="fa fa-home" aria-hidden="true"></i></a>  Measurement Conversion</h2><hr>
+          </div>
+          <div class="d-flex flex-wrap text-center">
+            <!-- <form>
+              <div class="form-group">
+                <input type="email" class="form-control" id="email1" placeholder="Your email address...">
+              </div>
+              <div class="form-group">
+                <input type="password" class="form-control" id="password1" placeholder=  "Your password...">
+              </div> -->
```
Comment (dhaval-mistry-simform) [inline index.php]: Remove this code.

File: js/script.js [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+    to = $("#type").val();
+    data = { "type": to, "o1": opt1, "o2": opt2, "v1": val1 };
+
+    $.ajax({
+        method: "POST",
+        url: "/conversion.php",
+        dataType: "json",
+        data: data,
+        success: function (val) {
+            // debugger;
```
Comment (dhaval-mistry-simform) [inline js/script.js]: Remove this code.

File: js/script.js [hunk 24]
```diff
@@ -0,0 +1,22 @@
+$(".btn").click(function () {
+    opt1 = $("#opt1").val();
```
Comment (dhaval-mistry-simform) [inline js/script.js]: Provide variable is let, var, const.

File: mail.php [hunk 25]
```diff
@@ -0,0 +1,90 @@
+<?php
+ini_set('display_errors', 1);
+ini_set('display_startup_errors', 1);
+error_reporting(E_ALL);
+if(isset($_POST['submit'])){
+    $to=$_POST['to'];
+    // echo"here";
```
Comment (dhaval-mistry-simform) [inline mail.php]: Remove this code.

File: mail.php [hunk 26]
```diff
@@ -0,0 +1,90 @@
...
+    $body=$_POST['body'];
+    $from="From: hekahekameka@gmail.com";
+        echo "her11dsrf";
+
+    if(mail($to, $sub, $body, $from)){
+        echo "mail sent to: $to, having :$sub//$body ,,, by $from";
+    }
+}
+
+// phpinfo();
```
Comment (dhaval-mistry-simform) [inline mail.php]: Remove this code.

File: mail.php [hunk 27]
```diff
@@ -0,0 +1,90 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <title>Converter</title>
+  <!-- Bootstrap CSS -->
+  <link rel='stylesheet' href='https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.2/css/bootstrap.min.css'>
+  <!-- Font Awesome CSS -->
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css"/>
+  <!-- Style CSS -->
+  <link rel="stylesheet" href="css/style.css">
+  <!-- Demo CSS -->
+  <!-- <link rel="stylesheet" href="css/demo.css"> -->
```
Comment (dhaval-mistry-simform) [inline mail.php]: Remove this code.

File: numbers.php [hunk 28]
```diff
@@ -0,0 +1,116 @@
+<?php
+
+if(isset($_POST['submit'])){
```
Comment (dhaval-mistry-simform) [inline numbers.php]: Please format this

File: numbers.php [hunk 29]
```diff
@@ -0,0 +1,116 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <title>Converter</title>
+  <!-- Bootstrap CSS -->
+  <link rel='stylesheet' href='https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.2/css/bootstrap.min.css'>
+  <!-- Font Awesome CSS -->
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css"/>
+  <!-- Style CSS -->
+  <link rel="stylesheet" href="css/style.css">
+  <!-- Demo CSS -->
+  <!-- <link rel="stylesheet" href="css/demo.css"> -->
```
Comment (dhaval-mistry-simform) [inline numbers.php]: Remove this code.

File: speed.php [hunk 30]
```diff
@@ -0,0 +1,85 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <title>Converter</title>
+  <!-- Bootstrap CSS -->
+  <link rel='stylesheet' href='https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.2/css/bootstrap.min.css'>
+  <!-- Font Awesome CSS -->
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/all.min.css"/>
+  <!-- Style CSS -->
+  <link rel="stylesheet" href="css/style.css">
+  <!-- Demo CSS -->
+  <!-- <link rel="stylesheet" href="css/demo.css"> -->
```
Comment (dhaval-mistry-simform) [inline speed.php]: Remove this code.

**Review by dhavalmistrysimform — CHANGES_REQUESTED** [review]
Please work on the given comments

File: conversion.php [hunk 31]
```diff
@@ -0,0 +1,195 @@
...
+    "Kmps" => 3600,
+    "Knot" => 1.852
+);
+
+// this fuction converts any length to meter and then from meter to users selected unit
+function len($val, $from, $to)
+{
+    $meter = toMeter($val, $from);
+    echo fromMeter($meter, $to);
+}
```
Comment (dhavalmistrysimform) [inline conversion.php:50]: Add new line after this

File: conversion.php [hunk 32]
```diff
@@ -0,0 +1,195 @@
...
+// this fuction converts any length to meter 
+function toMeter($val, $from)
+{
+    return $val * meter[$from];
+}
```
Comment (dhavalmistrysimform) [inline conversion.php:55]: Add new line after this

File: conversion.php [hunk 33]
```diff
@@ -0,0 +1,195 @@
...
+{
+    return $val / meter[$to];
+}
+
+// this fuction converts any volume to liters and then from liters to users selected unit
+function vol($val, $from, $to)
+{
+    $liter = toLtr($val, $from);
+    echo fromLtr($liter, $to);
+}
```
Comment (dhavalmistrysimform) [inline conversion.php:67]: Add new line after this

File: conversion.php [hunk 34]
```diff
@@ -0,0 +1,195 @@
...
+function speed($val, $from, $to)
+{
+    $kilo=toKMPH($val, $from);
+    echo fromKMPH($kilo, $to);
+}
+// this fuction converts any speed to KMPH 
+function toKMPH($val, $from)
+{
+    return $val * kmph[$from];
+}
```
Comment (dhavalmistrysimform) [inline conversion.php:160]: General Comment: Add new line after every function.

File: conversion.php [hunk 35]
```diff
@@ -0,0 +1,195 @@
...
+    return $val / kmph[$to];
+}
+
+// this if statement is executed on ajax call it checks from where the ajax is called and redirects execution to desired function
+if(isset($_POST['type'])){
+    $value = $_POST["v1"];
+    $from = $_POST["o1"];
+    $to = $_POST["o2"];
+
+    switch($_POST['type']){
```
Comment (dhaval-mistry-simform) [inline conversion.php]: Add space after switch and before opening curly braces

File: currency.php [hunk 36]
```diff
@@ -0,0 +1,92 @@
...
+if(isset($_POST['submit'])){
+  $from=$_POST['opt1'];
+  $to=$_POST['opt2'];
+  $value=$_POST['val1'];
+  $reqUrl = "https://api.exchangerate-api.com/v4/latest/$from";
+  $response_json = file_get_contents($reqUrl);
+
+  // Continuing if we got a result
+  if(false !== $responseJson) {
+      // Try/catch for json_decode operation
```
Comment (dhaval-mistry-simform) [inline currency.php]: Remove this unwanted code.

File: mail.php [hunk 37]
```diff
@@ -0,0 +1,64 @@
+<?php
+if(isset($_POST['submit'])){
```
Comment (dhaval-mistry-simform) [inline mail.php]: Format this document perfectly. Make sure you provide required spacing.
for eg. **if (isset($_POST['submit'])) {**

**Review by dhavalmistrysimform — CHANGES_REQUESTED** [review]
Please work on given comments

File: conversion.php [hunk 38]
```diff
@@ -0,0 +1,207 @@
...
+    case 'Kelvin':
+      return $val + 273.15;
+      break;
+  }
+}
+
+// this fuction converts any area to Square Meters and then from Square Meters to users selected unit
+function area($val, $from, $to)
+{
+    $meter=toSqMeter($val, $from);
```
Comment (dhaval-mistry-simform) [inline conversion.php]: Add spacing between =

File: conversion.php [hunk 39]
```diff
@@ -0,0 +1,207 @@
...
+// this fuction converts Square Meters to users selected unit
+function fromSqMeter($val, $to)
+{
+    return $val / pow(meter[$to], 2);
+}
+
+// this fuction converts any speed to KMPH and then from KMPH to users selected unit
+function speed($val, $from, $to)
+{
+    $kilo=toKMPH($val, $from);
```
Comment (dhaval-mistry-simform) [inline conversion.php]: Add spacing between =

File: mail.php [hunk 40]
```diff
@@ -0,0 +1,64 @@
+<?php
+if (isset($_POST['submit'])) {
+    $to=$_POST['to'];
```
Comment (dhaval-mistry-simform) [inline mail.php]: Add spacing between =

**Review by dhavalmistrysimform — CHANGES_REQUESTED** [review]
Please work on given comments

File: currency.php [hunk 41]
```diff
@@ -0,0 +1,90 @@
+<?php
+if(isset($_POST['submit'])){
+  $from=$_POST['opt1'];
+  $to=$_POST['opt2'];
+  $value=$_POST['val1'];
```
Comment (dhaval-mistry-simform) [inline currency.php]: Add spacing between =

File: js/script.js [hunk 42]
```diff
@@ -0,0 +1,21 @@
+$(".btn").click(function () {
```
Comment (dhaval-mistry-simform) [inline js/script.js]: Place this code inside document.ready or IIFE

File: numbers.php [hunk 43]
```diff
@@ -0,0 +1,101 @@
+<?php
+
+if(isset($_POST['submit'])){
+  $from=$_POST['opt1'];
+  $to=$_POST['opt2'];
+  $value=$_POST['val1'];
```
Comment (dhaval-mistry-simform) [inline numbers.php]: Please format this.

**Review by dhavalmistrysimform — CHANGES_REQUESTED** [review]
Please work on the given comments.

File: numbers.php [hunk 44]
```diff
@@ -0,0 +1,101 @@
+<?php
+
+if( isset($_POST['submit']) ){
```
Comment (dhaval-mistry-simform) [inline numbers.php]: it should be like this if (isset($_POST['submit'])) {

**Review by dhavalmistrysimform — CHANGES_REQUESTED** [review]
Work on given comments

**Pull request merged**